### ISSUE TITLE
httpCors: swallows thrown error if event looks like an API Gateway event

### ISSUE THREAD

**Issue by Wytrykus** [middyjs/middy#802]
**Describe the bug**
If the event passed to the handler function looks like an API Gateway event (e.g. it has a non-falsy `httpMethod` field), then the `httpCors` middleware makes an async "middy-fied" handler function resolve _even though_ an internal error (without http status code!) was thrown inside the handler and not handled by any middleware.

**To Reproduce**
How to reproduce the behaviour:

**A) handler code without the `httpCors` middleware**:

```typescript
import middy from '@middy/core'
import httpErrorHandler from '@middy/http-error-handler'

const innerHandler = async (): Promise<void> => {
  // pretend that some error is thrown
  await Promise.reject(new Error('something went wrong'))
}

export const handler = middy(innerHandler)
  .use(httpErrorHandler({ logger: undefined }))
```
This unit test runs successfully (using `jest` as testing framework):
```typescript
import { Context } from 'aws-lambda'
import { handler } from './handler-throwing-error'

describe('handler-throwing-error', () => {
  it('throws error if the inner handler throws error', async () => {
    const expectedError = new Error('something went wrong')

    await expect(
      handler(
        {
          httpMethod: 'GET',
          // some other stuff
        },
        {} as Context
      )
    ).rejects.toThrow(expectedError)  // expectation fulfilled
  })
})
```

**B) handler code using the `httpCors` middleware**:
```typescript
import middy from '@middy/core'
import httpCors from '@middy/http-cors'
import httpErrorHandler from '@middy/http-error-handler'

const innerHandler = async (): Promise<void> => {
  // pretend that some error is thrown
  await Promise.reject(new Error('something went wrong'))
}

export const handler = middy(innerHandler)
  .use(httpCors())
  .use(httpErrorHandler({ logger: undefined }))
```
After this change, the unit test fails with the message:
```
    Received promise resolved instead of rejected
    Resolved to value: {"headers": {"Access-Control-Allow-Origin": "*"}}
```

**Expected behaviour**

The `httpCors` middleware should not mess with internal server errors, because maybe the intended behaviour of the AWS lambda function is to _fail_ completely if there is an internal server error, instead of resolving.

In other words, the above unit test should run successfully with and without `httpCors` middleware.

**Environment (please complete the following information):**
 - Node.js: 14
 - Middy: 2.5.7

**Additional context**
The described issue is not reproducible if the event passed to the handler function does not look like an API Gateway event, because then the `httpCors` middleware is inactive, which is obviously due to this condition:
https://github.com/middyjs/middy/blob/main/packages/http-cors/index.js#L37

**Comment by willfarrell** [comment]
Thanks for reporting with such detail! This should already be resolved in v3-alpha. We reversed the `onError` call order to match `after` and update middleware to only apply changes to `response` iff the error has already been handled. We hope to have a beta out next month, with a full release early Apr.

**Comment by mcrstudio** [comment]
This is a painful issue and I can't upgrade to v3 because I start getting a myriad of other issues.

**Comment by willfarrell** [comment]
@mcrstudio Can you open a new issue with with more details on what you're running into?

**Comment by serverlesspolska** [comment]
I've been struggling with the same error the whole morning.
```
  "dependencies": {
    "@middy/core": "^2.5.3",
    "@middy/http-cors": "^2.5.3",
    "@middy/http-error-handler": "^2.5.3",
    "@middy/http-json-body-parser": "^2.5.3",
    "@middy/validator": "^2.5.3",
```

Lambda `handler`:
```JavaScript
module.exports.handler = middy(handler)
  .use(httpJsonBodyParser())
  .use(validator({ inputSchema }))
  .use(httpErrorHandler())
  .use(cors())
  .use(stringifyResponse())

const stringifyResponse = () => ({
  after: async (request) => {
    request.response = {
      statusCode: 200,
      body: JSON.stringify(request.response)
    }
  }
})
```

I've got following error thrown by AWS DynamoDB SDK (v2)
```
INFO {
  "message": "Missing required key 'TableName' in params",
  "code": "MissingRequiredParameter",
  "time": "2022-04-21T12:27:54.750Z"
}

INFO MissingRequiredParameter: Missing required key 'TableName' in params
    at ParamValidator.fail (/var/runtime/node_modules/aws-sdk/lib/param_validator.js:50:37)
    at ParamValidator.validateStructure (/var/runtime/node_modules/aws-sdk/lib/param_validator.js:61:14)
    at ParamValidator.validateMember (/var/runtime/node_modules/aws-sdk/lib/param_validator.js:88:21)
    at ParamValidator.validate (/var/runtime/node_modules/aws-sdk/lib/param_validator.js:34:10)
    at Request.VALIDATE_PARAMETERS (/var/runtime/node_modules/aws-sdk/lib/event_listeners.js:132:42)
    at Request.callListeners (/var/runtime/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at callNextListener (/var/runtime/node_modules/aws-sdk/lib/sequential_executor.js:96:12)
    at /var/runtime/node_modules/aws-sdk/lib/event_listeners.js:86:9
    at finish (/var/runtime/node_modules/aws-sdk/lib/config.js:396:7)
    at /var/runtime/node_modules/aws-sdk/lib/config.js:414:9 {
  code: 'MissingRequiredParameter',
  time: 2022-04-21T12:27:54.750Z
}
```

And it results in the following response from API Gateway **instead of the Error**:
```
HTTP/1.1 200 OK
Date: Thu, 21 Apr 2022 12:27:58 GMT
Content-Type: application/json
Content-Length: 47
Connection: close
Apigw-Requestid: Q7kYRiTVoAMEVhg=

{
  "headers": {
    "Access-Control-Allow-Origin": "*"
  }
}
```
